lib/npo.src.js

```diff
@@ -12,12 +12,59 @@
 	/*jshint validthis:true */
 	"use strict";
 
-	var builtInProp, cycle, scheduling_queue,
+	var builtInProp, cycle, scheduling_queue, nextShouldNotEmit = false,
 		ToString = Object.prototype.toString,
 		timer = (typeof setImmediate != "undefined") ?
 			function timer(fn) { return setImmediate(fn); } :
-			setTimeout
+			setTimeout,
+        //Adapted from [when](https://github.com/cujojs/when/blob/master/lib/makePromise.js)
+        emit = (function () {
+			function noop(){}
+			if (typeof process === 'object' && typeof process.emit === 'function') {
+				return function $$emit(type, data) {
+					if (type === 'unhandledRejection') {
+						return process.emit(type, data.reason, data.promise);
+					}
+					return process.emit(type, data);
+				}
+			}
+			if (typeof CustomEvent === 'function') {
+				return (function (CustomEvent, noop, self) {
+					var hasCustomEvent = false;
+					try {
+						var ev = new CustomEvent('unhandledRejection');
+						hasCustomEvent = ev instanceof CustomEvent;
+					} catch(e){}
+       	       
+					return hasCustomEvent ? function (type, data) {
+						var evt;
+						type = type.toLowerCase();
+						if (type === 'unhandledrejection') {
+							evt = new CustomEvent(type, {
+								detail: {
+									reason: data.reason,
+									promise: data.promise
+								},
+								bubbles: false,
+								cancelable: true
+							});
+						} else {
+							evt = new CustomEvent(type, {
+								detail: {
+									promise: data
+								},
+								bubbles: false,
+								cancelable: true
+							});
+						}
+						return !self.dispatchEvent(evt);
+					} : noop;
+				})(CustomEvent, noop, (typeof self === 'object' && typeof self.dispatchEvent === 'function') ? self : window);
+			}
+			return noop;
+		})()
```

Comment:
this looks like a decent implementation, and good that it was taken from another established lib. I'm bummed about its length. Any idea how much gzip'd size this adds to the minified distribution file? can you think of anything else trade-off wise that we could do to shorten this, even if it meant less support?